### PR TITLE
Add and refactor events

### PR DESCRIPTION
**Pull request by BillyDM**
I made a new pull request that only modifies the event structure.

### CODE REVIEW

File: examples/open_window.rs [hunk 1]
```diff
@@ -24,42 +24,11 @@ impl WindowHandler for MyProgram {
...
-            Event::WindowFocus => {
-                println!("Window focused");
-            }
-            Event::WindowUnfocus => {
-                println!("Window unfocused");
-            }
-            Event::WillClose => {
-                println!("Window will close");
-            }
+            Event::Interval(delta_time) => println!("Update interval, delta time: {}", delta_time),
```
Comment (wrl) [inline examples/open_window.rs:27]: What is the `Interval` event?

Comment (BillyDM) [reply examples/open_window.rs:27]: Its when the user requests to have a message called in periodic intervals. For example, I may want to process the DSP of my visualizer 30 frames a second or so to save performance, then request a redraw. The plugin will still run at the native frame-rate.

Comment (BillyDM) [reply examples/open_window.rs:27]: If you have something else in mind for this, I'll remove it.

**Comment by wrl** [comment]
Also, please rebase on top of master.

File: examples/open_window.rs (hunk 1 above)
Comment (wrl) [reply examples/open_window.rs:27]: Please remove it for now. While we will need this in the future for timers/animations or whatever, I'm not certain that this is the best way to handle it.

**Comment by BillyDM** [comment]
Also, I am getting this error:
```error[E0658]: use of unstable library feature 'option_zip'
  --> src/x11/window.rs:96:14
   |
96 |             .zip(xcb_connection.atoms.wm_delete_window)
   |              ^^^
   |
   = note: see issue #70086 <https://github.com/rust-lang/rust/issues/70086> for more information```

**Comment by wrl** [comment]
> Also, I am getting this error:
> 
> ```
> error[E0658]: use of unstable library feature 'option_zip'
>   --> src/x11/window.rs:96:14
>    |
> 96 |             .zip(xcb_connection.atoms.wm_delete_window)
>    |              ^^^
>    |
>    = note: see issue #70086 <https://github.com/rust-lang/rust/issues/70086> for more information```
> ```

`Option::zip()` is stable as of 1.46.0. You may need to update your toolchain.

**Comment by BillyDM** [comment]
Oh right

**Pull request merged**